src/ert/storage/local_experiment.py

```diff
@@ -334,7 +334,67 @@ def observation_keys(self) -> List[str]:
     def response_key_to_response_type(self) -> Dict[str, str]:
         mapping = {}
         for config in self.response_configuration.values():
-            for key in config.keys:
+            for key in config.keys if config.has_finalized_keys else []:
                 mapping[key] = config.response_type
 
         return mapping
+
+    @cached_property
+    def response_type_to_response_keys(self) -> Dict[str, List[str]]:
+        result: Dict[str, List[str]] = {}
+
+        for response_key, response_type in self.response_key_to_response_type.items():
+            if response_type not in result:
+                result[response_type] = []
+
+            result[response_type].append(response_key)
+
+        for keys in result.values():
+            keys.sort()
+
+        return result
+
+    def _has_finalized_response_keys(self, response_type: str) -> bool:
+        responses_configuration = self.response_configuration
+        if response_type not in responses_configuration:
+            raise KeyError(
+                f"Response type {response_type} does not exist in current responses.json"
+            )
+
+        return responses_configuration[response_type].has_finalized_keys
+
+    def _update_response_keys(
```

Comment:
What happens to wildcards in `keys` here?

Reply:
They will "disappear", and be replace with the actual first saved keys. So we expect no wildcards whenever `.has_finalized_keys` is `True`